bundles/org.openhab.binding.salus/src/main/java/org/openhab/binding/salus/internal/handler/CloudBridgeHandler.java

```diff
@@ -0,0 +1,269 @@
+import org.openhab.core.thing.binding.BaseBridgeHandler;
+import org.openhab.core.types.Command;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.github.benmanes.caffeine.cache.Caffeine;
+import com.github.benmanes.caffeine.cache.LoadingCache;
+
+/**
+ * @author Martin Grześlowski - Initial contribution
+ */
+@NonNullByDefault
+public final class CloudBridgeHandler extends BaseBridgeHandler implements CloudApi {
+    private Logger logger = LoggerFactory.getLogger(CloudBridgeHandler.class.getName());
+    private final HttpClientFactory httpClientFactory;
+    @NonNullByDefault({})
+    private LoadingCache<String, SortedSet<DeviceProperty<?>>> devicePropertiesCache;
+    @Nullable
+    private SalusApi salusApi;
+    @Nullable
+    private ScheduledFuture<?> scheduledFuture;
+
+    public CloudBridgeHandler(Bridge bridge, HttpClientFactory httpClientFactory) {
+        super(bridge);
+        this.httpClientFactory = httpClientFactory;
+    }
+
+    @Override
+    public void initialize() {
+        try {
+            internalInitialize();
+        } catch (Exception ex) {
+            updateStatus(OFFLINE, CONFIGURATION_ERROR, "Cannot start server! " + ex.getMessage());
+        }
+    }
+
+    private void internalInitialize() {
+        var config = this.getConfigAs(CloudBridgeConfig.class);
+        if (!config.isValid()) {
+            updateStatus(OFFLINE, CONFIGURATION_ERROR, "Username or password is not valid");
+            return;
+        }
+        var httpClient = new JettyHttpClient(httpClientFactory.getCommonHttpClient());
+        var localSalusApi = salusApi = new SalusApi(config.getUsername(), config.getPassword().toCharArray(),
+                config.getUrl(), httpClient, GsonMapper.INSTANCE);
+        logger = LoggerFactory
+                .getLogger(CloudBridgeHandler.class.getName() + "[" + config.getUsername().replace(".", "_") + "]");
+
+        var scheduledPool = ThreadPoolManager.getScheduledPool("Salus");
+        scheduledPool.schedule(() -> tryConnectToCloud(localSalusApi), 1, MICROSECONDS);
+
+        this.devicePropertiesCache = Caffeine.newBuilder().maximumSize(10_000)
+                .expireAfterWrite(Duration.ofSeconds(config.getPropertiesRefreshInterval()))
+                .refreshAfterWrite(Duration.ofSeconds(config.getPropertiesRefreshInterval()))
+                .build(this::loadPropertiesForDevice);
+        this.scheduledFuture = scheduledPool.scheduleWithFixedDelay(this::refreshCloudDevices,
+                config.getRefreshInterval() * 2, config.getRefreshInterval(), SECONDS);
+
+        // done
+        updateStatus(ONLINE);
+    }
+
+    private void tryConnectToCloud(SalusApi localSalusApi) {
+        try {
+            localSalusApi.findDevices();
+        } catch (Exception ex) {
+            updateStatus(OFFLINE, COMMUNICATION_ERROR,
+                    "Cannot connect to Salus Cloud! Probably username/password mismatch! " + ex.getMessage());
```

Comment:
updated code

Reply:
Did you push the changes? Have not noticed it yet.

Reply:
Check the whole method

Reply:
If you refer to SalusApi.findDevices and the underlying login, i do see different exceptions, but those are not reflected in this method. Here only the generic `Exception` is catched and we would like to see that only checked exceptions are catched: Extend custom exceptions from `Exception`

Reply:
`tryConnectToCloud` is run in `ScheduledExecutorService`. I need to catch all exceptions that might happen because if this fail the binding will not be working (it has some problems with connection), but it will not be seen anywhere, because the executor service will swallow it

Reply:
Catch them at the place of the actual API call and route them to a custom exception. This custom exception can be consumed by the BridgeHandler to do whatever it needs like setting the ThingStatusDetail to Badd username/password, even with i18n posibilities.